Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.Vessel.swift

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extension AddCustomDynamicEntityDataSourceView {
+    /// A marine vessel that can be decoded from the vessel JSON.
+    struct Vessel {
+        /// The location of the vessel.
+        let geometry: Geometry
+        /// The attributes of the vessel.
+        let attributes: [String: Any]
+    }
+}
+extension AddCustomDynamicEntityDataSourceView.Vessel: Decodable {
+    /// A geometry that gives the location of the vessel.
+    struct Geometry: Decodable {
+        /// The x coordinate of the geometry.
+        let x: Double
+        /// The y coordinate of the geometry.
+        let y: Double
+    }
+    
+    private enum CodingKeys: CodingKey {
+        case geometry
+        case attributes
+    }
+    
+    init(from decoder: Decoder) throws {
+        /// The attributes that define meta data for the vessel.
+        struct Attributes: Decodable {
+            enum CodingKeys: String, CodingKey { // swiftlint:disable:this nesting
+                case mmsi = "MMSI"
+                case sog = "SOG"
+                case cog = "COG"
+                case vesselName = "VesselName"
+                case callSign = "CallSign"
```

Comment:
Since this is a _really_ private type there's not a whole lot of benefit to following the API design guidelines regarding naming. You could just use the JSON key names as the vessel properties names, removing the need for this enum.

Reply:
True but I don't see a huge benefit to removing the enum. I would prefer to keep the camel case conventions of variable names intact.